Fix PCP Invalid operation code

cas-- opened the pull request:
While testing the client with miniupnpd 2.2.1 on OpenWRT the following
error occurred:

     PCP(Invalid response: Invalid operation code: No discriminant in enum OperationCode matches the value 65)

The RFC describes a Response bit followed by 7-bit Opcode

```
 0 1 2 3 4 5 6 7
|R|   Opcode    |
```

With the original bit shifting would change the Opcode value so instead
we need to mask the response bit.

This is my first Rust PR so hopefully the changes and tests follow best practice...

cas-- committed:
refactor: cargo fmt

cas-- committed:
fix(pcp): mask instead of bitshift pcp opcodes

While testing the client with miniupnpd 2.2.1 on OpenWRT the following
error occurred:

   PCP(Invalid response: Invalid operation code: No discriminant in enum OperationCode matches the value 65)

The RFC describes a Response bit followed by 7-bit Opcode

0 1 2 3 4 5 6 7
|R|   Opcode    |

With the original bit shifting would change the Opcode value so instead
we need to mask the response bit.

cas-- committed:
feat(pcp): raise error if Response bit is unset

ryco117 commented:
Awesome, thanks for the PR! I'll give the changes a more thorough reading and testing later but seems good at first glance.

Admittedly, my own hardware only supports NAT-PMP, so I was just going off my best interpretation of the RFC for my PCP implementation, so I really appreciate the feedback.

ryco117 commented:
Ah, one thing I'll ask is that you remove `opcode_to_request(op)`  and replace its usage with just `op as u8` since the enum is defined such that the MSb will be `0`.  The reason I was shifting was because of my confusion about the LSb in the RFC's notation.

cas-- committed:
refactor(pcp): remove redundant opcode_to_request

The OperationCode enum MSb should always be zero so masking to unset R
bit is redundant.

cas-- commented:
Done, I was actually in two-minds about keeping or removing that function... :+1:

ryco117 commented:
Thanks again @cas-- ! I may reformat the tests slightly to my own personal tastes, but then I'll try to get an update to the crates.io before the end of the weekend so that the public lib can be fixed.

Cheers
